insights/core/serde.py

```diff
@@ -95,15 +95,37 @@ def deserialize(data, root=None):
 
 
 def marshal(v, root=None, pool=None):
+    data = {
+        'errors': [],
+        'results': None
+    }
```

Comment:
This `data` makes the code a bit complex, why not just return a tuple from the inner `call_serializer` function?
e.g.
```python
        try:
            return func(value), None
        except Exception:
            return None, traceback.format_exc()
```

Reply:
@xiangce
It is to keep consistent with before. When there is one result for the spec, we should set it to the result, instead of a list.
As we discussed,  I found we'll return different types for the `marshal` function in this way. And I have to check the values based on the returned type. 
And in the test file, when we test the `marchal` function, after getting the result, we also need to check the returned value type. So I'm wondering if it is good to return different types for one function.

Reply:
@xiangce updated as discussed, please help to review, thanks.